### PR TITLE
Request for Access changes and fixing Roles and Permissions

### PR DESCRIPTION
**Pull request by sankalp0o**
Fixing https://github.com/postmanlabs/postman-docs/issues/1573 and https://github.com/postmanlabs/postman-docs/issues/1813

### CODE REVIEW

**Comment by sankalp0o** [comment]
For more context, this is the release spec for which the changes were made (apart from the docs fixes) -https://postmanlabs.atlassian.net/wiki/spaces/COL/pages/1602881848/Collaboration+0.3+In+Beta

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    url: "/docs/postman/collections/sharing-collections/"
+  - type: link
+    name: "Collection Roles"
+    url: "/docs/postman-pro/managing-pro/roles-and-permissions/#collection-roles"
+
+warning: false
+
+---
+
+You can request access to a team member's personal collection or a private collection in your team if you have the collection link. To do so, follow these steps
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: I'd just skip the last sentence here "to do so..."

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Go to the collection link
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Am I right in thinking you paste the url into a browser? Can we say that in case people think it's something they do inside the app..?

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+2. Click on **Request Access**
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: "Click" is fine instead of "Click on"

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+![https://assets.postman.com/postman-docs/access-blocked.png](https://assets.postman.com/postman-docs/access-blocked.png)
+
+3. Select a person to send the request to, select a role ([read more](/docs/postman-pro/managing-pro/roles-and-permissions/#collection-roles) about collection roles), and select one of your workspace to add the collection to once you are given access.
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Instead of the "read more" in brackets, I'd just make the text "role" link to that page. 

This looks like more than one step, maybe break it up? Think there's a typo re "workspace" - assuming should be "workspace".

Comment (SueSmith) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: Could we also explain who the person to send the request to should be?

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+![https://assets.postman.com/postman-docs/request-access.png](https://assets.postman.com/postman-docs/request-access.png)
+
+4. Complete your request by clicking on **Request Access**
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Again "clicking" is fine without "on".. Missing a period at the end here I think.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+This will trigger an email to the person you requested access from. When they approve your request, you will be notified by an email. Until your request is approved or gets expired (after 15 days), you won't be able to make a request on the same collection.
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: "notified by email" is fine without "an"

I'd reword slightly "You will be able to make requests in the collection when your access is approved (and until it expires in 15 days)."

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+</br>
+
+## Request Editor Role on a Collection
+
+There might be collections that you can view but not edit. These collections have a 'lock' icon next to the collection name.
+
+<img src="https://assets.postman.com/postman-docs/collection-lock.png" width="300px">
+
+If you want to request the Editor role on this collection, you can do so by following these steps
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: "You can request the Editor role on this collection:"

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+1. Click on the ellipsis icon (...) next to the collection name in the sidebar to open the following menu and click on **Manage Roles**
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Again if we can lose "on" after both clicks here. I'd lose "to open the following menu". Also needs a period at the end.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+<img src="https://assets.postman.com/postman-docs/collection-context-menu.png" width="300px">
+
+2. Click on **Request Access** in the modal
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Same again, I'd lose "on" and add a period at the end.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+![https://assets.postman.com/postman-docs/manage-roles-blocked.png](https://assets.postman.com/postman-docs/manage-roles-blocked.png)
+
+3. Select a person to send the request to, select the role you wanted ([read more](/docs/postman-pro/managing-pro/roles-and-permissions/#collection-roles) about collection roles). In this case, this will be the "Editor" role.
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Same note as above re linking to role doc inline etc.

Comment (SueSmith) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: Last sentence I'd just make "Select __Editor__ from the __Role__ dropdown list."

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+![https://assets.postman.com/postman-docs/request-additional-access.png](https://assets.postman.com/postman-docs/request-additional-access.png)
+
+4. Complete your request by clicking on **Request Access**
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Same again for "on" and period at end.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+This will trigger an email to the person you requested access from. When they approve your request, you will be notified by an email. Until your request is approved or gets expired (after 15 days), you won't be able to make a request on the same collection.
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Same wording as above.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+3. Select a person to send the request to, select a role ([read more](/docs/postman-pro/managing-pro/roles-and-permissions/#collection-roles) about collection roles), and select one of your workspace to add the collection to once you are given access.
+![https://assets.postman.com/postman-docs/request-access.png](https://assets.postman.com/postman-docs/request-access.png)
+
+4. Complete your request by clicking on **Request Access**
+
+This will trigger an email to the person you requested access from. When they approve your request, you will be notified by an email. Until your request is approved or gets expired (after 15 days), you won't be able to make a request on the same collection.
+
+</br>
+
+## Request Editor Role on a Collection
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Trying to shift to sentence case and "-ing" for subheadings, so "Requesting editor role on a collection"

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+3. Select a person to send the request to, select the role you wanted ([read more](/docs/postman-pro/managing-pro/roles-and-permissions/#collection-roles) about collection roles). In this case, this will be the "Editor" role.
+![https://assets.postman.com/postman-docs/request-additional-access.png](https://assets.postman.com/postman-docs/request-additional-access.png)
+
+4. Complete your request by clicking on **Request Access**
+
+This will trigger an email to the person you requested access from. When they approve your request, you will be notified by an email. Until your request is approved or gets expired (after 15 days), you won't be able to make a request on the same collection.
+
+</br>
+
+## Approving Requests for Access
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: Approving requests for access

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+
+Your team members can request access from you in the following cases:
+
+1. If your team members have the link to your personal collection, they won't be able to view it but they will be able to see the name and request access to it.
+
+2. For collections that are shared, a team member who has the Viewer role will be able to request additional role from you if you have the Editor role.
+
+In both these cases, you will receive an email with the following details
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: I'd lose the start of this sentence and just go for "You will receive an email detailing the request for access."

File: src/pages/docs/postman/collections/requesting-access-to-collections.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+![https://assets.postman.com/postman-docs/request-access-mail.png](https://assets.postman.com/postman-docs/request-access-mail.png)
+
+If you want to approve this request, you can click on the **Approve** button and the request will be approved. If you do not wish to give access, you can ignore the request. A request for access expires in 15 days.
```
Comment (SueSmith) [inline src/pages/docs/postman/collections/requesting-access-to-collections.md]: I'd lose the "on" again..

Does the request for access expire as well as the granted access?

File: src/pages/docs/postman/collections/requesting-access-to-collections.md (hunk 2 above)
Comment (sankalp0o) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: You can either paste the URL into the browser (if someone gave it to you) or in some cases you might be following a link in the app.
How do we cover both these cases without being specific about how to do this? To be honest, this is not a step that the user intentionally takes, maybe we reword it to something like "If you encounter this state where you don't have access to a collection you're trying to view...".

File: src/pages/docs/postman/collections/requesting-access-to-collections.md (hunk 4 above)
Comment (sankalp0o) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: Umm, if you take a look at the interface, it's just 2 input fields so didn't want to split it.
The person to send the request to would in 99% of the cases just be 1 person, so don't think there is a decision involved.

File: src/pages/docs/postman/collections/requesting-access-to-collections.md (hunk 6 above)
Comment (sankalp0o) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: What I meant was that "Your request would automatically expire in 15 days. You would be able to create another one after the request expires"

File: src/pages/docs/postman/collections/requesting-access-to-collections.md (hunk 16 above)
Comment (sankalp0o) [reply src/pages/docs/postman/collections/requesting-access-to-collections.md]: If not granted, the request expires. The access doesn't have any expiry.

**Comment by sankalp0o** [comment]
@SueSmith I've made most of the changes and resolved those comments. Can you take it from here please? :) 

I've tried to give context for the remaining comments, but let me know if you need more information.
